src/directives/shared.ts

```diff
@@ -219,6 +219,40 @@ export const DIRECTIVES: Directives = {
                 return 'fulfilled';
             }
 
+            // Catch all for edge cases like a "invalidClaim" status
+            return 'unfulfilled';
+        },
+    },
+    storageMappings: {
+        token: 'dd1319b2-e72b-421c-ad2b-082352569bb1',
+        queryFilter: (queryBuilder) => {
+            return queryBuilder;
+        },
+        generateUserClaim: (args: any[]) => {
+            return {
+                addStore: args[0],
+                catalogPrefix: args[1],
+            };
+        },
+        calculateStatus: (appliedDirective?) => {
+            // If there is no directive to check it is unfulfilled
+            if (!appliedDirective || isEmpty(appliedDirective)) {
+                return 'unfulfilled';
+            }
+
+            // If directive already queued and no claim is there we can just use that directive again
+            if (
+                appliedDirective.job_status.type === 'queued' &&
+                !appliedDirective.user_claims
+            ) {
+                return 'in progress';
+            }
```

Comment:
Can this actually happen for this directive?

Reply:
Haha no, the second condition should be removed for this directive at the very least.